src/client/shadows/dynamicshadowsrender.cpp

```diff
@@ -129,6 +149,37 @@ f32 ShadowRenderer::getMaxShadowFar() const
 	return 0.0f;
 }
 
+bool ShadowRenderer::addPointLight(v3f position, v3s16 light_node_pos)
+{
+	MapNode light_node = m_client->getEnv().getClientMap().getNode(light_node_pos);
+
+	const NodeDefManager* ndef_mgr = m_client->getNodeDefManager();
+	const ContentFeatures& cf = ndef_mgr->get(light_node.getContent());
+
+	// This node has to contain param1 value
+	if (cf.param_type != CPT_LIGHT)
+		return false;
+
+	m_pointlight_list.emplace_back(m_shadow_map_texture_size,
+			// farPlane  -- I don`t know how to calculate this parameter.
```

Comment:
Does anybody know how to calculate a maximum distance on which a point light can extend depending on its `light_source` intensivity parameter? Somebody suggested to go me to light.cpp and use `set_light_data(float gamma)`, but I don't understand what it calculates and how to use those returned parameters for that.